### DISCUSSION TITLE
Custom properties of the current user's profile

### DISCUSSION THREAD

**Discussion by pnicosia80** [oqtane/oqtane.framework#4905] [Q&A] [answered]
Hi all,
I need to access from my client side module to the custom properties of the current user's profile, how can I do it?

Thanks in advance!

**Comment by Raceeend** [comment]
The profile info is stored in the settings of the User. An example is the code of UserProfile in the Oqtane Client module (in the folder Oqtane.Client Modules/Admin/UserProfile).

**Comment by leigh-pointer** [comment, marked as answer]
@pnicosia80 
you will need to inject a couple of services

```
@inject IUserService UserService
@inject IProfileService ProfileService
```

in code declare
```
private List<Profile> profiles;
private Dictionary<string, string> settings;
```

in your OnInitializedAsync() method you can poputlate the declared variables

```
profiles = await ProfileService.GetProfilesAsync(ModuleState.SiteId);
settings = await SettingService.GetUserSettingsAsync(PageState.User.UserId);
```

You can extract the Profile values from the settings using this function

```
    private string GetProfileValue(string SettingName, string DefaultValue)
    {
        string value = SettingService.GetSetting(settings, SettingName, DefaultValue);
        if (value.Contains("]"))
        {
            value = value.Substring(value.IndexOf("]") + 1);
        }
        return value;
    }
```

**Reply by pnicosia80** [reply]
Hi, thanks! problem solved.

**Comment by sbwalker** [comment]
@pnicosia80 @leigh-pointer there is a much simpler way to access the current user's profile settings:

PageState.User.Settings()

The information is already loaded and available - you don't need to call any services.
